src/11-brushy-cli/data.ts

```diff
@@ -0,0 +1,203 @@
+        }
+        rarity: attributionClass {
+          name
+        }
+        medium: mediumType {
+          name
+        }
+        materials: medium
+        price: saleMessage
+      }
+
+      fragment user on User {
+        name
+        email
+        follows {
+          genesConnection(first: 20) {
+            edges {
+              node {
+                name
+              }
+            }
+          }
+          artistsConnection(first: 20) {
+            edges {
+              node {
+                name
+                genes {
+                  name
+                }
+              }
+            }
+          }
+        }
+        myCollectionArtworksConnection(first: $numMyCollectionArtworks) {
+          edges {
+            node {
+              ...artwork
+            }
+          }
+        }
+      }
+    `,
+
+    variables: {
+      artworkID,
+      userID,
+      numMyCollectionArtworks,
+    },
+
+    headers: {
+      "X-Access-Token": process.env.ADMIN_ACCESS_TOKEN!,
+    },
+  })
+
+  // console.log(JSON.stringify(data, null, 2))
+
+  function cleanUp(names: { name: string }[]) {
+    return names.map((x) => x.name).join("; ")
+  }
+
+  // clean up user - follows
+  const followGenes = extractNodes(data.user.follows.genesConnection)
+  data.user.followedCategories = cleanUp(followGenes)
+  delete data.user.follows.genesConnection
+
+  const followArtists = extractNodes(data.user.follows.artistsConnection)
+  data.user.followedArtists = followArtists
+  delete data.user.follows.artistsConnection
+
+  data.user.followedArtists.forEach((artist) => {
+    artist.categories = cleanUp(artist.genes)
+    delete artist.genes
+  })
+  delete data.user.follows
+
+  // clean up user - MyC
+  data.user.collectedArtworks = extractNodes(
+    data.user.myCollectionArtworksConnection
+  )
+  data.user.collectedArtworks.forEach((artwork) => {
+    artwork.artist.categories = cleanUp(artwork.artist.categories)
+    delete artwork.slug
+  })
+  delete data.user.myCollectionArtworksConnection
+
+  // clean up artwork
+  if (data.artwork.artist) {
+    data.artwork.artist.categories = cleanUp(
+      data.artwork.artist.categories || []
+    )
+  }
+
+  // clean up partner
+  data.artwork.partner.categories = cleanUp(data.artwork.partner.categories)
+
+  const { user: userData, artwork: artworkArtistPartnerData } = data
+  const { artist, partner, ...artwork } = artworkArtistPartnerData
+  const { collectedArtworks, ...user } = userData
+
+  // enrich artwork with partner applied categories
+  const { genes: pacs } = await gravity(
+    `partner/${partner.slug}/artwork/${artwork.slug}/genome`,
+    {},
+    { "X-Access-Token": process.env.ADMIN_ACCESS_TOKEN! }
+  )
+  artwork.categories = Object.keys(pacs).join("; ")
+
+  // console.log({ artwork, partner, artist, user, collectedArtworks })
+  return { artwork, partner, artist, user, collectedArtworks }
+}
+
+export async function getPartnerAppliedGene(name: string) {
```

Comment:
Don't have to resolve in this PR, but just wonder out loud if we should extract out these functions into the `tools` folder as this data will likely be useful for a variety of use cases?

Reply:
Ah interesting point. Probably so, I'll fold that into my tools cleanup PR (which is looking more like next week at this point)